.github/actions/build-ruby/index.js

```diff
@@ -107,24 +116,104 @@ async function setupRubyEnvironment(rubyVersion) {
   core.exportVariable('SERIALIZE', 1)
 }
 
-async function setupRubyEnvironmentAfterBuild(rubyVersion) {
+// Sets up any options at the bundler level so that when gems that 
+// need specific settings are installed, their specific flags are relayed.
+async function configureBundleOptions(rubyVersion) {
   if (!usesOldOpenSsl(rubyVersion)) { return }
 
+  const openSslPath = rubyOpenSslPath(rubyVersion);
+  
+  // https://stackoverflow.com/questions/30834421/error-when-trying-to-install-app-with-mysql2-gem
+  await exec.exec('bundle', [
+    'config', '--global', 'build.mysql2',
+      `"--with-ldflags=-L${openSslPath}/lib"`,
+      `"--with-cppflags=-I${openSslPath}/include"`
+  ]);
+}
+
+// prepends the given value to the environment variable
+function prependEnv(envName, envValue, divider=' ') {
+  let existingValue = process.env[envName];
+  if (existingValue) {
+    envValue += `${divider}${existingValue}`
+  }
+  core.exportVariable(envName, envValue);
+}
+
+// The older Rubies also need older MySQL that was built against the older OpenSSL libraries.
+// Otherwise mysql adapter will segfault in Ruby because it attempts to dynamically link 
+// to the 1.1 series while Ruby links against the 1.0 series.
+async function downgradeMySQL() {
+  core.startGroup(`Downgrade MySQL`)
+
+  const pkgDir = `${process.env.HOME}/packages`
+  const pkgOption = `--directory-prefix=${pkgDir}/`
+  const mirrorUrl = 'https://mirrors.mediatemple.net/debian-security/pool/updates/main/m/mysql-5.5'
+
+
+  // executes the following all in parallel  
+  let a = exec.exec('sudo', ['apt-get', 'remove', 'mysql-client']);
+  let b = exec.exec('wget', [pkgOption, `${mirrorUrl}/libmysqlclient18_5.5.62-0%2Bdeb8u1_amd64.deb`]);
+  let c = exec.exec('wget', [pkgOption, `${mirrorUrl}/libmysqlclient-dev_5.5.62-0%2Bdeb8u1_amd64.deb`]);
+
+  // wait for the parallel processes to finish
+  await a;
+  await b;
+  await c;
+
+  // executes serially
+  await exec.exec('sudo', ['dpkg', '-i', `${pkgDir}/libmysqlclient18_5.5.62-0+deb8u1_amd64.deb`]);
+  await exec.exec('sudo', ['dpkg', '-i', `${pkgDir}/libmysqlclient-dev_5.5.62-0+deb8u1_amd64.deb`]);
+
+  core.endGroup()
+}
+
+// mySQL (and others) must be downgraded for EOL rubies for native extension
+// gems to install correctly and against the right openSSL libraries.
+async function downgradeSystemPackages(rubyVersion) {
+  if (!usesOldOpenSsl(rubyVersion)) { return }
+
+  await downgradeMySQL();
+}
+
+// any settings needed in all Ruby environments from EOL'd rubies to current
+async function setupAllRubyEnvironments() {
```

Comment:
Yes.  I started to yank it altogether, but decided it was a good one to keep because my JS expertise will wilt over time.  Thus this No Op quickly frames where to embellish for our future selves.

Reply:
Okay, seems reasonable. Thanks!